datalad/runner/runner.py

```diff
@@ -211,8 +212,10 @@ def run(self,
         lgr.debug("Finished %r with status %s", cmd, results['code'])
 
         # make it such that we always blow if a protocol did not report
-        # a return code at all
-        if results.get('code', True) not in [0, None]:
```

Comment:
@mih - this is the `True` default which puzzled me and I switched to `None` but may be incorrectly. May be you recall motivation in 9551b91929f6ee8e48a92fb343cdfdfbc05cef22 -- e.g. "when there is no `code` here we really not treat it as `None` but want to get `None` from somewhere deeper within"?

Reply:
I am not sure what this wants to say

> when there is no code here we really not treat it as None but want to get None from somewhere deeper within

but what this conditional did was stated in the still present comment:

> make it such that we always blow if a protocol did not report a return code at all

Because in the linked commit it established:

> The result is now always a dict that is expected to have keys 'code', 'stdout', 'stderr', matching the semantics of the CommandError exception.

The changed code seems to achieve the same behavior: raise if there is no `code` in the results. So LGTM.

Reply:
I can live with LGTM without dwelling on the rest. Cheers!